### DISCUSSION TITLE
Is there a way to include extra OSM tag to edges?

### DISCUSSION THREAD

**Discussion by caspg** [valhalla/valhalla#4755] [General]
I would like to include orignal OSM surface tag inside each edge. Is that possible right now without changing source?

**Comment by nilsnolde** [comment]
nope, but you can get the way ID from `/locate` or so, send that to overpass api and get the original surface tag if any. we do infer quite a bit from context for surface I think.

**Reply by caspg** [reply]
I would like to get surface stats for a cycling routes which will contain quite a lot of ways. In this case hitting overpass would be quite costly I guess.

**Reply by nilsnolde** [reply]
I was never a heavy overpass user, but I'd assume you can pass it multiple way IDs in one query and get back the surface tags for them. might need your own instance bcs of rate limits/fair usage, but I don't see another way without adding code (and we wouldn't want that in master I think).

**Reply by caspg** [reply]
Got ya. Thanks for the answer.
